### PR TITLE
Fix broken links in md files

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

- Add action to check broken links.
- Fix broken links

### Reason for changes

Existed broken links and no any test to check it.

### Related tickets

131357

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
Example of output on develop:
```
File: CONTRIBUTING.md:27 • Link: ./tests/data • Error: Path does not exist
File: README.md:44 • Link: ./docs/compression_algorithms/Quantization.md#mixed_precision_quantization • Error: Not found fragment
File: docs/Usage.md:306 • Link: ./accuracy_aware_model_training/AdaptiveCompressionTraining.md • Error: Path does not exist
File: docs/Usage.md:317 • Link: ./accuracy_aware_model_training/AdaptiveCompressionTraining.md • Error: Path does not exist
File: docs/Usage.md:381 • Link: ./accuracy_aware_model_training/AdaptiveCompressionTraining.md • Error: Path does not exist
File: docs/Usage.md:383 • Link: ../../examples/torch/classification/main.py • Error: Path is not within git repository
File: docs/Usage.md:383 • Link: ../../examples/torch/classification/configs/pruning/resnet18_cifar10_accuracy_aware.json • Error: Path is not within git repository
File: examples/tensorflow/object_detection/README.md:9 • Link: https://github.com/tensorflow/models/tree/master/official/vision/detection • Error: 404: Not Found
File: examples/tensorflow/segmentation/README.md:9 • Link: https://github.com/tensorflow/models/tree/master/official/vision/detection • Error: 404: Not Found
File: nncf/experimental/torch/sparsity/movement/MovementSparsity.md:81 • Link: (https://arxiv.org/pdf/2005.07683.pdf • Error: Path does not exist
File: nncf/experimental/torch/sparsity/movement/MovementSparsity.md:82 • Link: (https://arxiv.org/pdf/2109.04838.pdf • Error: Path does not exist
File: tests/onnx/README.md:25 • Link: benchmarking/README.md#prepare-models • Error: Not found fragment
File: tests/onnx/README.md:26 • Link: benchmarking/README.md#prepare-models • Error: Not found fragment
File: tests/onnx/benchmarking/README.md:25 • Link: https://github.com/onnx/models/blob/main/vision/classification/alexnet/model/bvlcalexnet-12.onnx • Error: 404: Not Found
...
```

**Review by alexsu52 — APPROVED** [review]

**Review by kshpv — APPROVED** [review]

File: .github/workflows/pre-commit-linters.yml [hunk 1]
```diff
@@ -19,3 +19,8 @@ jobs:
       run: make install-pre-commit
     - name: Run pre-commit linter suite
       run: make pre-commit
+  md-dead-link-check:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: AlexanderDokuchaev/md-dead-link-check@0.4
```
Comment (ljaljushkin) [inline .github/workflows/pre-commit-linters.yml:26]: is it implemented in another repo?

Comment (AlexanderDokuchaev) [reply .github/workflows/pre-commit-linters.yml:26]: yes

File: tests/onnx/README.md [hunk 2]
```diff
@@ -22,8 +22,8 @@ We provide two types of tests.
 
     You should give three arguments to run this test.
 
-    1. `--model-dir`: The directory path which includes ONNX Model ZOO models (.onnx files). See [#prepare-models](benchmarking/README.md#prepare-models) for details.
-    2. `--data-dir`: The directory path which includes datasets (ImageNet2012, COCO, Cityscapes, and VOC) [#prepare-models](benchmarking/README.md#prepare-models).
+    1. `--model-dir`: The directory path which includes ONNX Model ZOO models (.onnx files). See [#prepare-models](benchmarking/README.md#benchmark-for-onnx-models-vision) for details.
+    2. `--data-dir`: The directory path which includes datasets (ImageNet2012, COCO, Cityscapes, and VOC) [#prepare-models](benchmarking/README.md#1-prepare-dataset).
```
Comment (ljaljushkin) [inline tests/onnx/README.md:26]:
```suggestion
    2. `--data-dir`: The directory path which includes datasets (ImageNet2012, COCO, Cityscapes, and VOC) [#prepare-dataset](benchmarking/README.md#1-prepare-dataset).
```

**Pull request merged**